gems/aws-sdk-core/lib/seahorse/client/plugins/raise_response_errors.rb

```diff
@@ -21,7 +21,7 @@ def call(context)
 
         def add_handlers(handlers, config)
           if config.raise_response_errors
-            handlers.add(Handler, step: :validate, priority: 95)
+            handlers.add(Handler, step: :initialize, priority: 95)
```

Comment:
There is an issue with client side metrics because it rescues and re-raises errors, and it relies on this being where it's at. I found a solution that doesn't involve changing this, though I agree, we should ideally have this as late as possible.